synapse/handlers/pagination.py

```diff
@@ -373,12 +373,15 @@ async def get_messages(
                     # case "JOIN" would have been returned.
                     assert member_event_id
 
-                    leave_token = await self.store.get_topological_token_for_event(
+                    leave_token_str = await self.store.get_topological_token_for_event(
                         member_event_id
                     )
-                    if RoomStreamToken.parse(leave_token).topological < max_topo:
+                    leave_token = RoomStreamToken.parse(leave_token_str)
+                    assert leave_token.topological
```

Comment:
Should this be `is not None`?

Reply:
Also, surely we should be using if+exception here instead of `assert`? It doesn't look like a condition that's impossible to reach to me.

Reply:
`get_topological_token_for_event` should absolutely be returning a token which has a topological part. 

I'm not a huge fan of using if + exception for informing mypy about such assumptions, as it has the potential to be quite nosiy.

Reply:
> get_topological_token_for_event should absolutely be returning a token which has a topological part.

Oh, ignore me, I misread the code.